das_client/lib/sfera/src/mapper/sfera_model_mapper.dart

```diff
@@ -96,6 +105,46 @@ class SferaModelMapper {
     );
   }
 
+  static BaseData? _calculateCurrentPosition(List<BaseData> journeyData, List<SegmentProfileList> segmentProfilesLists,
+      RelatedTrainInformation? relatedTrainInformation, Journey? lastJourney) {
+    if (relatedTrainInformation == null ||
+        relatedTrainInformation.ownTrain.trainLocationInformation.positionSpeed == null) {
```

Comment:
Nitpick: `relatedTrainInformation.ownTrain.trainLocationInformation.positionSpeed` evtl. in Variable auslagern, da lang und 4x verwendet.